pkgs/build-support/fetchurl/default.nix

```diff
@@ -124,5 +134,12 @@ if (!hasHash) then throw "Specify hash for fetchurl fixed-output derivation: ${s
   # traffic, so don't do that.
   preferLocalBuild = true;
 
+  inherit fetchType;
+  fetchArguments = {
+     url = head urls;
+     hash = outputHash;
+     type = outputHashAlgo;
+  };
```

Comment:
I don't think this is necessary, since this information is already available via other attributes (`outputHash`, `urls`, etc.). To distinguish fetchurl calls from other fixed-output derivations, it should be enough to have some constant (pass-through) attribute (like `passthru.isFetchedURL`).

Also, doesn't the use of an attribute set make `derivation` fail?